### DISCUSSION TITLE
@click.outside doesn't work on template element in alpine.js

### DISCUSSION THREAD

**Discussion by mir161803** [alpinejs/alpine#3314] [7. Bugs (archived)]
I'm experimenting with `x-if` and `x-show` and found some interesting behavior.

if we create basic dropdown(hide/unhide) with `x-show="open"` and `@click.outside="open = false"` like below, the behavior result is as expected and my text will hide/unhide
```
...
// it works
<div x-show="open" @click.outside="open = false">
  my text
</div>
```

but when I try to replicate the same result with `x-if`, I couldn't get the same result (I mean hide and unhide, like we know that `x-if` will completely remove the element from dom and `x-show` will just give the `display : none`)

```
// it doesn't works
<template x-if="open" @click.outside="open = false">
  my test
</template>
```

**Comment by SimoTod** [comment]
The click outside should be on the first child not on the template tag if you want to replicate the x-show example.
X-if clones the first child of the template tag and adds it to the dom
